### PR TITLE
fix(api): move composite propagator to API

### LINKED ISSUE #1013: Propagators API
According to the [spec](https://sourcegraph.com/github.com/open-telemetry/opentelemetry-specification/-/blob/specification/context/api-propagators.md
)  the propagators https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/src/propagation/composite.rs should be in the API instead of the SDK 

Context Conversation: https://cloud-native.slack.com/archives/C03GDP0H023/p1679426866872239

### PR DESCRIPTION
**Pull request by TommyCpp**
Fixes #1013
Design discussion issue (if applicable) #

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry/src/propagation/mod.rs [hunk 1]
```diff
@@ -1,170 +1,30 @@
...
-//! Deserializing must set `is_remote` to true on the returned
-//! `SpanContext`.
+//! Currently, the following `Propagator` types are supported:
+//! -  [`TextMapPropagator`], inject values into and extracts values from carriers as string key/value pairs
 //!
-//! ## Binary Format
+//! A binary Propagator type will be added in
+//! the future, See [tracking issues](https://github.com/open-telemetry/opentelemetry-specification/issues/437)).
 //!
-//! `BinaryFormat` is a formatter to serialize and deserialize a value
```
Comment (hdost) [inline opentelemetry/src/propagation/mod.rs:15]: Do we want to remove all this documentation?

**Review by hdost — APPROVED** [review]
Looks good for the most part, just one question.

File: opentelemetry/src/propagation/mod.rs (hunk 1 above)
Comment (TommyCpp) [reply opentelemetry/src/propagation/mod.rs:15]: I think it mostly came from old specification which can be misleading. We can add stuff like `BinaryPropagator` back once we implement it

**Comment by TommyCpp** [comment]
@open-telemetry/rust-approvers Can I get another review?

File: opentelemetry/src/propagation/composite.rs [hunk 2]
```diff
@@ -107,31 +113,31 @@ impl TextMapPropagator for TextMapCompositePropagator {
     }
 }
 
-#[cfg(all(test, feature = "testing", feature = "trace"))]
```
Comment (jtescher) [inline opentelemetry/src/propagation/composite.rs:110]: do we want to keep the `trace` feature as this uses types from that module?

**Review by jtescher — APPROVED** [review]
Makes sense to move this 👍

**Comment by hdost** [comment]
> @open-telemetry/rust-approvers Can I get another review? (We discussed needing 2 review per PRs the other day)

We should ask to have this updated.  @jtescher do you have access to this or should we ask on the community repo?

**Pull request merged**